rest/model/admin.go

```diff
@@ -1496,9 +1473,206 @@ func (a *APIAWSConfig) ToService() (interface{}, error) {
+	}
+
+	return evergreen.AWSPodConfig{
+		Role:           utility.FromStringPtr(a.Role),
+		Region:         utility.FromStringPtr(a.Region),
+		ECS:            ecs,
+		SecretsManager: sm,
+	}, nil
+}
+
+// APIECSConfig represents configuration options for AWS ECS.
+type APIECSConfig struct {
+	Role                 *string               `json:"role"`
+	TaskDefinitionPrefix *string               `json:"task_definition_prefix"`
+	Clusters             []APIECSClusterConfig `json:"clusters"`
+}
+
+func (a *APIECSConfig) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case evergreen.ECSConfig:
+		a.TaskDefinitionPrefix = utility.ToStringPtr(v.TaskDefinitionPrefix)
+		for _, cluster := range v.Clusters {
+			var apiCluster APIECSClusterConfig
+			if err := apiCluster.BuildFromService(cluster); err != nil {
+				return errors.Wrap(err, "building API ECS cluster config")
+			}
+			a.Clusters = append(a.Clusters, apiCluster)
+		}
+		return nil
+	default:
+		return errors.Errorf("%T is not a supported type", h)
+	}
+}
+
+func (a *APIECSConfig) ToService() (interface{}, error) {
+	if a == nil {
+		return nil, nil
+	}
+	var clusters []evergreen.ECSClusterConfig
+	for _, apiCluster := range a.Clusters {
+		i, err := apiCluster.ToService()
+		if err != nil {
+			return nil, errors.Wrap(err, "building ECS cluster config")
+		}
+		cluster, ok := i.(evergreen.ECSClusterConfig)
+		if !ok {
+			return nil, errors.Errorf("programmatic error: unexpected type %T for ECS cluster config", i)
+		}
+		clusters = append(clusters, cluster)
+	}
+	return evergreen.ECSConfig{
+		TaskDefinitionPrefix: utility.FromStringPtr(a.TaskDefinitionPrefix),
+		Clusters:             clusters,
+	}, nil
+}
+
+// APIECSClusterConfig represents configuration options for a cluster in AWS ECS.
+type APIECSClusterConfig struct {
+	Name     *string `json:"name"`
+	Platform *string `json:"platform"`
+}
+
+func (a *APIECSClusterConfig) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case evergreen.ECSClusterConfig:
+		a.Name = utility.ToStringPtr(v.Name)
+		a.Platform = utility.ToStringPtr(string(v.Platform))
+		return nil
+	default:
+		return errors.Errorf("%T is not a supported type", h)
+	}
+}
+
+func (a *APIECSClusterConfig) ToService() (interface{}, error) {
+	if a == nil {
+		return nil, nil
+	}
+	return evergreen.ECSClusterConfig{
+		Name:     utility.FromStringPtr(a.Name),
+		Platform: evergreen.PodPlatform(utility.FromStringPtr(a.Platform)),
+	}, nil
+}
+
+// APISecretsManagerConfig represents configuration options for AWS Secrets
+// Manager.
+type APISecretsManagerConfig struct {
+	SecretPrefix *string `json:"secret_prefix"`
+	Region       *string `json:"region"`
+}
+
+func (a *APISecretsManagerConfig) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case evergreen.SecretsManagerConfig:
+		a.SecretPrefix = utility.ToStringPtr(v.SecretPrefix)
```

Comment:
Should we also copy the region?

Reply:
Sorry, I forgot to remove this. I deleted the region (which isn't used).